tests/distributed/test_distributed_sampling.py

```diff
@@ -0,0 +1,87 @@
+import dgl
+import unittest
+import os
+from dgl.data import CitationGraphDataset
+from dgl.distributed.sampling import sample_neighbors
+from dgl.distributed import partition_graph, load_partition, GraphPartitionBook
+import sys
+import multiprocessing as mp
+import numpy as np
+import backend as F
+import time
+from utils import get_local_usable_addr
+from pathlib import Path
+
+from dgl.distributed import DistGraphServer, DistGraph
+
+def myexcepthook(exctype, value, traceback):
+    for p in mp.active_children():
+        p.terminate()
+    # raise Exception("11111111111")
+
+
+
+def start_server(rank, tmpdir):
+    import dgl
+    g = DistGraphServer(rank, "rpc_sampling_ip_config.txt", 1, "test_sampling",
+                        tmpdir / 'test_sampling.json')
+    g.start()
+
+def start_client(rank, tmpdir):
+    import dgl
+    g = DistGraph("rpc_sampling_ip_config.txt", "test_sampling")
+    print("Pre sample")
+    print(g.number_of_nodes())
+    results = sample_neighbors(g, [0, 10, 99, 66, 1024, 2008], 3)
+    print("after sample")
+    dgl.distributed.shutdown_servers()
+    dgl.distributed.finalize_client()
+    return results
+
+
+@unittest.skipIf(os.name == 'nt', reason='Do not support windows yet')
+@unittest.skipIf(dgl.backend.backend_name != 'PyTorch', reason='Only support pytorch for now')
+def test_rpc_sampling(tmpdir):
+    num_server = 3
+    ip_config = open("rpc_sampling_ip_config.txt", "w")
+    ip_addr = get_local_usable_addr()
+    ip_config.write('127.0.0.1 20060 1\n')
+    ip_config.write('127.0.0.1 20070 1\n')
+    ip_config.write('127.0.0.1 20080 1\n')
+    ip_config.close()
+    # sys.excepthook = myexcepthook
+    # partition graph
+    g = CitationGraphDataset("cora")[0]
+    g.readonly()
+    print(g.idtype)
+    num_parts = num_server
+    num_hops = 1
+
+    partition_graph(g, 'test_sampling', num_parts, tmpdir,
+                    num_hops=num_hops, part_method='metis', reshuffle=False)
```

Comment:
please add a comment that this will be tested with range partition as well in the future.

Reply:
How to test it with RangePartitionBook. And should RangePartitionBook & GraphPartitionBook inherits an abstract class such as `PartitionBook`?

Reply:
it can be inherited from the same class. but does it matter in Python?

Reply:
as for how to test RangePartitionBook, you can use 'orig_id' to get the node Id in the original graph. something like below should work.
```python
sampled_graph.ndata['orig_id'] = dist_graph.ndata['orig_id'][sampled_graph.ndata[dgl.NID]]
src, dst = sampled_graph.edges()
src = sampled_graph.ndata['orig_id'][src]
dst = sampled_graph.ndata['orig_id'][dst]
assert np.all(F.asnumpy(g.has_edges_between(src, dst).bool()))
```

Reply:
> it can be inherited from the same class. but does it matter in Python?

Inheritance make it easy to understand that they have the same interface. (without documentation)